doc-content/drools-docs/src/main/asciidoc/DecisionEngine/cep-modes-con.adoc

```diff
@@ -0,0 +1,69 @@
+[id='cep-modes-con_{context}']
+
+= Event processing modes in the {DECISION_ENGINE}
+
+The {DECISION_ENGINE} runs in either cloud mode or stream mode. In cloud mode, the {DECISION_ENGINE} processes facts as facts with no temporal constraints, independent of time, and in no particular order. In stream mode, the {DECISION_ENGINE} processes facts as events with strong temporal constraints, in real time or near real time. Stream mode uses synchronization to make event processing possible in {PRODUCT}.
+
+Cloud mode::
+Cloud mode is the default operating mode of the {DECISION_ENGINE}. In cloud mode, the {DECISION_ENGINE} treats events as an unordered cloud. Events still have time stamps, but the {DECISION_ENGINE} running in cloud mode cannot draw relevance from the time stamp because cloud mode ignores the present time. This mode uses the rule constraints to find the matching tuples to activate and execute rules.
++
+--
+Cloud mode does not impose any kind of additional requirements on facts. However, because the {DECISION_ENGINE} in this mode has no concept of time, it cannot use temporal features such as sliding windows or automatic life-cycle management. In cloud mode, events must be explicitly retracted when they are no longer needed.
+
+The following requirements are not imposed in cloud mode:
+
+* No clock synchronization because the {DECISION_ENGINE} has no notion of time
```

Comment:
Oh, this reminds me that I completely forgot to give you a disclaimer on this doc. I use anthropomorphism absolutely everywhere. I tried really hard to avoid it, but the problem is that a lot of the language needs to be directly about the decision engine and what it does and how it behaves. And because it is indeed an AI system, it does do, know, think, in that sense. Wherever I could avoid it, I have, and where avoiding it made it less clear, I just stuck to the anthropomorphism.

In this case, although I'm not a fan and agree with you, I don't have a better phrase for this. To say "measurement of time" is okay, but not quite right still. It's more than just not measuring time; it's that the engine doesn't even consider time, time isn't a thing. I thought "has no consideration of time" but that sounds a bit clunky. What I have is the clearest to me so far, to my mind, but I'll definitely think on this for an alternative if I can.

Reply:
Good points, and I agree, anthropomorphism is generally needed for clarity in this doc.